### PR TITLE
Unify RGB and RGBW commands

### PR DESCRIPTION
**Pull request by drashna**
## Description

* Unified RGB and RGBW API using `LED_TYPE`
* `rgblight.c` now clears W on RGBW LEDs

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* Closes #2350

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: drivers/arm/ws2812.h [hunk 1]
```diff
@@ -14,4 +14,3 @@
  *         - Wait 50�s to reset the LEDs
```
Comment (fauxpark) [inline drivers/arm/ws2812.h]:
```suggestion
 *         - Wait 50µs to reset the LEDs
```

File: drivers/arm/ws2812.c [hunk 2]
```diff
@@ -84,17 +84,12 @@ void ws2812_setleds(LED_TYPE *ledarray, uint16_t leds) {
         sendByte(ledarray[i].g);
         sendByte(ledarray[i].r);
         sendByte(ledarray[i].b);
+#ifdef RGBW
+        sendByte(ledarray[i].b);
```
Comment (fauxpark) [inline drivers/arm/ws2812.c]:
```suggestion
        sendByte(ledarray[i].w);
```

File: drivers/avr/ws2812.h [hunk 3]
```diff
@@ -37,5 +37,13 @@
  *         - Wait 50�s to reset the LEDs
```
Comment (fauxpark) [inline drivers/avr/ws2812.h]:
```suggestion
 *         - Wait 50µs to reset the LEDs
```

File: common_features.mk [hunk 4]
```diff
@@ -265,7 +265,7 @@ ifneq ($(strip $(BACKLIGHT_ENABLE)), no)
     endif
 endif
 
-VALID_WS2812_DRIVER_TYPES := bitbang pwm spi i2c
+VALID_WS2812_DRIVER_TYPES := bitbang pwm spi i2c bb_twi
```
Comment (zvecr) [inline common_features.mk]: The direction here wont work correctly, as ledbbtwi does not actually implement the ws2812 driver interface.

ledbbtwi.c existing in avr seems fine, though its only being used by one keyboard so maybe a more generic "custom" option is better.

File: drivers/arm/ledbbtwi.c [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#include "ledbbtwi.h"
```
Comment (zvecr) [inline drivers/arm/ledbbtwi.c]: Better to remove (and the header) and let the build fail, as its not like we have the i2c ps2avrgb implementation copied to ARM.

File: drivers/avr/ws2812.h [hunk 6]
```diff
@@ -34,8 +34,16 @@
  * The functions will perform the following actions:
  *         - Set the data-out pin as output
  *         - Send out the LED data
- *         - Wait 50�s to reset the LEDs
+ *         - Wait 50µs to reset the LEDs
  */
 void ws2812_setleds(LED_TYPE *ledarray, uint16_t number_of_leds);
-void ws2812_setleds_pin(LED_TYPE *ledarray, uint16_t number_of_leds, uint8_t pinmask);
-void ws2812_setleds_rgbw(LED_TYPE *ledarray, uint16_t number_of_leds);
+
```
Comment (zvecr) [inline drivers/avr/ws2812.h]: This block should be reverted, as it bleeds out internal implementation, and instead rely on the forward deceleration in `drivers/avr/ws2812.c`

File: drivers/avr/ws2812_i2c.c [hunk 7]
```diff
@@ -21,11 +21,3 @@ void ws2812_setleds(LED_TYPE *ledarray, uint16_t leds) {
 
     i2c_transmit(WS2812_ADDRESS, (uint8_t *)ledarray, sizeof(LED_TYPE) * leds, WS2812_TIMEOUT);
 }
-
-// Setleds for SK6812RGBW
-void ws2812_setleds_rgbw(LED_TYPE *ledarray, uint16_t leds) {
-// not supported - for now error out if its enabled
-#ifdef RGBW
```
Comment (zvecr) [inline drivers/avr/ws2812_i2c.c]: This logic has to be maintained, as the ps2avrGB boards wont handle the extra bytes at all. Now the _rgbw function has been remove, the error logic should exist at the top of the file.

File: common_features.mk (hunk 4 above)
Comment (drashna) [reply common_features.mk]: Well, this does work.

Specifically, what the bb_twi does is add the ledbbtwi file, and add a define to enable it. 

And the problem with "custom" is that the core ledbbtwi stuff doesn't reside in the ws2812.c, but rgblight.

File: drivers/arm/ledbbtwi.c (hunk 5 above)
Comment (drashna) [reply drivers/arm/ledbbtwi.c]: Can do

Comment (drashna) [reply drivers/arm/ledbbtwi.c]: Actually, will rip out entire ledbbtwi separation for now, in favor of RGBW support

**Comment by drashna** [comment]
Will basically completely redo entire PR to only include RGBW support.

**Comment by zvecr** [comment]
Okeydoke, I have some local bodges/ideas for the ergodox i2c driver side of things, give me a ping (discord?) when you want to work on that area!

**Comment by drashna** [comment]
Okay, PR gutted and redone. 

Now, this only unifies RGB and RGBW functions into a single call.  No need for a separate RGBW command.

File: drivers/arm/ws2812.c [hunk 8]
```diff
@@ -84,17 +84,12 @@ void ws2812_setleds(LED_TYPE *ledarray, uint16_t leds) {
         sendByte(ledarray[i].g);
         sendByte(ledarray[i].r);
         sendByte(ledarray[i].b);
-    }
+#ifdef RGBW
+        sendByte(ledarray[i].w);
+#endif
+}
```
Comment (fauxpark) [inline drivers/arm/ws2812.c]:
```suggestion
    }
```

Comment (drashna) [reply drivers/arm/ws2812.c]: That was already there .... :D

That said, ran `qmk cformat` on the files I touched.  Should be cleaner, now.

**Pull request merged**